core/src/main/java/com/lafaspot/imapnio/async/internal/ImapAsyncSessionImpl.java

```diff
@@ -317,6 +329,20 @@ public void handleIdleEvent(@Nonnull final IdleStateEvent idleEvent) {
             return;
 
         } else if (serverResponse.isTagged()) {
+            if (currentCmd instanceof CompressCommand && serverResponse.isOK()) {
```

Comment:
we should avoid looking for command type

Reply:
Done by having a method in ImapRequest interface isCompressionRequested().